Dynamic work-cards load and work popup details for mobile and desktop devices.

lmunozm1702 opened the pull request:
## Implements dynamic HTML generation for work cards:

- Create a data structure for storing the content of each card.
- Replace the static HTML code on index.html with a dynamic generation from javascript functions.
- Uses the DOM manipulation methods to build the nodes with the content.

## Implement card detail popup:

- For both, mobile and desktop devices.
- Reuse of all code for both, making changes on classes with javascript and CSS media querys.

lmunozm1702 committed:
Change fixed text for a javascript dynamic load

Update index.js to implement the dynamic construction of the CARDS section from a data structure to the html, using javascript.

Update index.html to delete the fixed contents in the Work Cards Section.

Update .gitignore file to add some utilitarian files.

lmunozm1702 committed:
Add mobile popup open with dynamic content.

Update style.css, index.js and index.html to code the open the popup with detail info for an work card.

Add static-data.js to save the structure with the cards data.

HevarKas committed:
driver:hevar navigator: luis - fixing style

lmunozm1702 committed:
Add popup closing for mobile

Updates index.js to implement the close action for mobile popup.

lmunozm1702 committed:
Add work details popup for desktop devices.

Update style.css and index.js for implement the card details popup for desktop devices.

Add grey-line.png on /assets/images folder.

lmunozm1702 committed:
Solve linter errors.

lmunozm1702 committed:
Fix linters errors

lmunozm1702 committed:
solve linter errors

lmunozm1702 committed:
Solve JS linter errors

lmunozm1702 committed:
Solve JS linter errors

lmunozm1702 committed:
Solve linter errors

lmunozm1702 committed:
Solve linter errors.

lmunozm1702 committed:
Change import/export module to default

AmaduKamara reviewed (changes requested):
# Status: Required Changes ♻️♻️♻️ 

Hi @lmunozm1702,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

## Required Changes ♻️
_Check the comments under the review._

- [x] I would be nice if you could work on the badges in this design so they match with the original design. Currently, when the modal opens, the texts in the badges get wrapped. I will suggest you reduce the with of the wrapper that is holding the paragraph text that describes the project on the left of the badges so the badges can have breathing space.
- [x] Also, you could reduce the width of the modal and the image in the modal. The size of the image is making the modal window too big and does not have a good experience. You can reduce the image size so the modal on desktop screens will look nice. The mobile design looks great 👍 

**Your Design**
![image](https://user-images.githubusercontent.com/50941074/195611335-c60f1ca3-eedd-4716-a755-681c0a9c9ee3.png)

**The original design**
![image](https://user-images.githubusercontent.com/50941074/195611520-442b7af6-8672-4c89-8042-437c5a95dd1f.png)


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

lmunozm1702 committed:
Solve Code Review Issues

Update style.css to solve:

- No wrapping into the languajes badges.
- wrap on the badges container to have 3 badges per row.

- Reduce the with of the popup, I set 2 sizes for popup at 768 and 1440 px with.

- removes some animations and transformations because increase the size of the main images and looks like over the popup.

lmunozm1702 committed:
linter error

lmunozm1702 commented:
Hi!, both issues are solved.

- For the first I set nowap to the text and force the wrap into the container.

![image](https://user-images.githubusercontent.com/53025062/195644578-ca162f5f-1998-4af6-a557-b7d1f8c2ab9e.png)

- For second, I set popup sizes for 768 and 1440 px.

SajjadAhmad14 reviewed (approved):
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Luis, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).
